src/java.base/share/classes/sun/security/ssl/SSLContextImpl.java

```diff
@@ -71,9 +71,29 @@ public abstract class SSLContextImpl extends SSLContextSpi {
     private volatile StatusResponseManager statusResponseManager;
 
     private final ReentrantLock contextLock = new ReentrantLock();
-    final HashMap<Integer,
-            SessionTicketExtension.StatelessKey> keyHashMap = new HashMap<>();
 
+    protected SessionTicketExtension.StatelessKey getKey(int id) {
+        return serverCache.getKey(id);
+    }
+
+    protected SessionTicketExtension.StatelessKey getKey() {
+        SessionTicketExtension.StatelessKey ssk = serverCache.getKey();
```

Comment:
I may change the serverCache from private to package private, and move the getKey() methods into SSLSessionContextImpl:
```
-    private final SSLSessionContextImpl serverCache;
+    final SSLSessionContextImpl serverCache;
```

Then the caller could use sslContext.serverCache.getKey() directly.  BTW, I may change the name from getKey to getStatelessKey() for readability.

Reply:
I did the refactoring as requested, but I don't like (and I don't think it is "good practice") to expose fields to other classes. And there's already a getter for `serverCache` in `SSLContextImpl` anyway (i.e. `SSLContextImpl::engineGetServerSessionContext`) so I decided to use that one instead and leave the field as private.